### PR TITLE
Created  Code Generator for React JS  in reference to issue #351

### PR DESCRIPTION
**Pull request by 2019UCP1350**
As the popularity of react js to build websites and web apps is increasing, therefore, there is a need for the hour to build codegen for reactjs.

 Here are the few websites which show the popularity of reactjs 
 https://ezeelive.com/reactjs-popularity/
 https://www.heliossolutions.co/blog/top-7-reasons-why-reactjs-development-is-so-popular/

Moreover, One of the major social media web-app Whatsapp is also built-in reactjs

### CODE REVIEW

**Comment by Sharath-Postman** [comment]
@abhijitkane if you could look into this.

original ref issue -- https://github.com/postmanlabs/gsoc/issues/36

**Comment by 2019UCP1350** [comment]
> @abhijitkane 
Could you have a look at my work?

**Comment by umeshp7** [comment]
@2019UCP1350 We already have [nodejs-axios](https://github.com/postmanlabs/postman-code-generators/tree/develop/codegens/nodejs-axios). 
Could you summarize in a comment how different is this codegen from the that one?

**Comment by VShingala** [comment]
@2019UCP1350 As @umeshp7 suggested, we already have added support for reactjs (axios) framework. From what I gathered in PR, it also do it in axios right?

In that case, can you mention if existing code gen for reactjs works for you as expected and if not does this PR add any new features to it?

**Comment by 2019UCP1350** [comment]
> @2019UCP1350 As @umeshp7 suggested, we already have added support for nodejs (axios) framework. From what I gathered in PR, it also do it in axios right?
> 
> In that case, can you mention if existing code gen for axios works for you as expected and if not does this PR add any new features to it?

When I had made this PR, Someone has requested the codegen snippet for reactjs, and it was not there.
Moreover, I used the try, catch and async, await syntax rather than nested then and catch syntax making the code snippet look pleasant and cleaner.
Also, currently, only node js Axios support is there, and snippet generates code for node js rather than react js which uses import differently than node js.

**Comment by VShingala** [comment]
@2019UCP1350 Makes sense! Thanks for the contribution and fast resonse. We'll evaluate these changes and see if we can accommodate them in Postman app codegen snippets.

**Review by dhwaneetbhatt — CHANGES_REQUESTED** [review]
@2019UCP1350 Rather than this being a different option altogether, I think passing it as an option to the original NodeJS-Axios generator would be good. We can keep it multi-select to be NodeJS or ReactJS, and it can generate any additional different code for ReactJS. Most of the code would remain the same. What are your thoughts on this?

We have recently upgraded the NodeJS-Axios generator with changes to produce async/await compatible code. Could you please update your fork from the latest upstream develop and see if this can work as part of the same generator?
